### PR TITLE
feat(api): list OAuth registrations a caller may authorize (INT-8)

### PR DESCRIPTION
**Pull request by erichare**
## Why

Registration configuration is operator-only (`LANGFLOW_CONNECTION_OAUTH_REGISTRATIONS`), and nothing exposed the ids that `POST /connections/{connection_id}/oauth/start` accepts, or the scope ceiling a start request must stay inside. A connection UI had to guess the registration id — and guessing the provider id fails on the naming convention our own docs use (`google-work`), so the authorize flow cannot be completed on a realistic configuration.

This is the gap the INT-8 connections UI hit (the demo console in `scratch/` works around it by reading the operator's env file directly, which a real client cannot do).

## What

`GET /api/v1/connections/oauth/registrations`, optionally filtered by `?provider=`:

```json
{
  "registrations": [
    {
      "id": "google-work",
      "provider": "google",
      "profile": "user",
      "context": "self_managed",
      "client_type": "confidential",
      "scopes": ["https://www.googleapis.com/auth/calendar.events.readonly"],
      "allowed_tenants": []
    }
  ]
}
```

- **Credential-free by construction.** The client id, client secret, private key and redirect URI stay on the server; the response model is `extra="forbid"` over an explicit field list, and a test asserts none of those strings appear in the body.
- **Only what consent could actually use.** A registration this deployment context would refuse (for example a `desktop` registration on a `self_managed` instance) is omitted rather than advertised, reusing `OAuthSettings.registration()` so availability logic stays in one place. Providers outside the integration policy's approved set are filtered the same way `GET /integrations` filters them.
- `OAuthSettings.registration_ids()` lists the configured names without validating them; every availability rule stays in `registration()`.

## Testing

`src/backend/tests/unit/api/v1/test_connection_registrations.py` — 8 tests: listing and exact payload shape, no client credentials in the response, provider filter, context-refused registration omitted, invalid registration omitted, empty configuration, authentication required, and an invalid provider filter rejected with 422.

Existing `test_connection_oauth.py` and `test_connections.py` still pass (30 tests), which covers the route not shadowing `GET /connections/oauth/{provider}/callback`.

## Follow-up

Pairs with #15165 (the `connection_ref` picker). Together they let a UI list connections for a field, and authorize a new one without the operator having to hand out registration ids out of band.

## Summary by CodeRabbit

* **New Features**
  * Added an authenticated endpoint for listing available OAuth registrations.
  * Supports filtering registrations by provider.
  * Displays non-sensitive registration details, including supported scopes and context.
  * Excludes unavailable, invalid, or policy-restricted registrations from results.

* **Documentation**
  * Added documentation describing the OAuth registration listing endpoint and its scope limitations.

### CODE REVIEW

File: src/backend/base/langflow/api/v1/connections.py [hunk 1]
```diff
@@ -399,6 +425,61 @@ async def start_connection_oauth(
     return OAuthStartResponse(authorization_url=url)
 
 
+@router.get("/oauth/registrations", response_model=OAuthRegistrationListRead)
```
[the remark this answers is not in the document]
Comment (erichare) [reply src/backend/base/langflow/api/v1/connections.py:428]: Agreed, fixed in b8228e236a: the handler sets `Cache-Control: no-store`, with a test asserting the header. The listing is filtered per caller, so it must not be shared even though it carries no credentials.

File: src/backend/tests/unit/api/v1/test_connection_registrations.py [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+    return configure
+
+
+async def test_lists_configured_registrations(client: AsyncClient, logged_in_headers, registrations):
+    _ = registrations
+    response = await client.get("api/v1/connections/oauth/registrations", headers=logged_in_headers)
+
+    assert response.status_code == 200, response.text
+    listed = response.json()["registrations"]
+    assert [entry["id"] for entry in listed] == ["google-work", "slack-bot"]
```
[the remark this answers is not in the document]
Comment (erichare) [reply src/backend/tests/unit/api/v1/test_connection_registrations.py:102]: Added in 72d321a562: `test_omits_providers_outside_the_integration_policy` configures Google and Slack, approves only Google through the policy bundle, and asserts Slack is omitted — covering `policy.allows_provider` independently of the deployment-context filter.
